book/src/solana-scripts.md

```diff
@@ -0,0 +1,108 @@
+# Solana Scripts
+
+This architecture describes how the Pipeline runtime supports scripts. 
+
+In Solana programs have absolute control over the state assigned to them.  The
+goal of this architecture is provide a simple way for on-chain programs to
+interact with other programs without violating the state ownership guarantees.
+
+Contract composition has led to multiple multi-million dollar bugs in Ethereum
+
+* https://www.parity.io/security-alert-2/
+
+## Scripts
+
+The purpose of scripts is to execute logic between instructions.
+
+For example, assignment of tokens to the winner based on an output of a game can
```

Comment:
You should mention that we can currently assign lamports, but not tokens, which are encapsulated by a program.